libraries/chain/include/eosio/chain/controller.hpp

```diff
@@ -141,12 +141,13 @@ namespace eosio::chain {
 
          /**
           * Starts a new pending block session upon which new transactions can be pushed.
+          * returns the trace for the on_block action
           */
-         void start_block( block_timestamp_type time,
-                           uint16_t confirm_block_count,
-                           const vector<digest_type>& new_protocol_feature_activations,
-                           block_status bs,
-                           const fc::time_point& deadline = fc::time_point::maximum() );
+         transaction_trace_ptr start_block( block_timestamp_type time,
+                                            uint16_t confirm_block_count,
+                                            const vector<digest_type>& new_protocol_feature_activations,
+                                            block_status bs,
+                                            const fc::time_point& deadline = fc::time_point::maximum() );
```

Comment:
Instead of modifying controller, seems like the tester should connect to `applied_transaction`  which would include the `onblock`.

Reply:
What is the drawback of having `start_block` returning the trace?

Reply:
Using the `applied_transaction` signal tests that the signal is called appropriately, would be nice to have some tests around that functionality. I don't have a strong rejection to having `start_block` return the `onblock` trace.

Reply:
Since we already get the transaction traces in libtester as a result of `control->push_transaction()`, I'd rather leave as-is. 
I think it seems appropriate for `start_block` to return the `onblock` trace.